File: BarberApp.Frontend/src/app/Models/UserConfig.ts
import { Checks } from './Checks';
import { AppColors } from './Enums/app-colors.enum';
export class UserConfig {
  darkmode = false;
  primaryColor: AppColors | string = 'main';
  secondaryColor = "#518fa3";
  altColor = "#f4ede8";
  fontSize = "16px";
  dueDate = "2023-02-24T04:59:08.260Z";
  checks?: Checks;
  pageTitles: any;


  constructor(params?: Partial<UserConfig>) {
    if (!params)
      return;

    Object.assign(this, params);
  }
}

export interface UserConfig{
  darkmode: boolean;
  primaryColor: AppColors | string,
  secondaryColor: string,
  altColor: string,
  fontSize: string,
  dueDate: string,
  checks?: Checks;
  pageTitles: any
}

File: BarberApp.Backend/BarberApp.DOMAIN/Models/UserConfig.cs
using MongoDB.Bson.Serialization.Attributes;
using MongoDB.Bson;
using System.Text.Json.Serialization;

namespace BarberApp.Domain.Models
{
    public class UserConfig
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        [JsonIgnore]
        [BsonElement("userConfigId")]
        public string UserConfigId { get; set; }
        [BsonElement("darkmode")]
        public bool Darkmode { get; set; }
        [BsonElement("primaryColor")]
        public string PrimaryColor { get; set; }
        [BsonElement("secondaryColor")]
        public string SecondaryColor { get; set; }
        [BsonElement("AltColor")]
        public string AltColor { get; set; }
        [BsonElement("fontSize")]
        public string FontSize { get; set; }
        [BsonElement("dueDate")]
        public DateTime DueDate { get; set; }
        [BsonElement("checks")]
        public Checks Checks { get; set; }
        [BsonElement("pageTitles")]
        public Dictionary<string, string> PageTitles { get; set; }
    }
}
